Add script (and ci job) to check if various version numbers are in sync

hikinggrass opened the pull request:
everest-framework has a few sub-projects that contain their own version numbers that should be in sync with the one defined in the toplevel CMakeLists.txt

- everest-framework: `CMakeLists.txt`
- everestrs: `package.json`
- everestpy: `__init__.py `and `everestpy.cpp`
- everestrs and everestrs-build: `Cargo.toml` files

hikinggrass committed:
Add script (and lint job) to check if various version numbers are in sync

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

andistorm commented on .github/workflows/build_and_test.yaml:
```diff
@@ -14,6 +14,19 @@ on:
+    name: Ensure versions are consistent
+    runs-on: ${{ inputs.runner || 'ubuntu-22.04' }}
+    defaults:
+      run:
+        working-directory: ./source
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          path: source
```
In the first moment written like this I would expect the checkout action to checkout in `./source/source` . So I would prefer setting working-directory explicitly for ` Run version consistency check` But this is just a matter of taste

hikinggrass replied:
I just oriented myself on the workflow(s) in everest-ci, but I'm open for alternatives :smile:

andistorm commented on .ci/build-kit/scripts/check_version_consistency.py:
```diff
@@ -0,0 +1,208 @@
+
+    if all(value == cpp_version for value in [everestjs_version, everestpy_python_version, everestpy_cpp_version, everestrs_version, everestrs_build_version]):
+        return 0
+    print('Version mismatch, you can fix this by calling this script with --fix which will apply the version defined in the toplevel CMakeLists.txt')
```
Above the version is named/listed as `C++ version`. Intuitive I wouldn't map the term `C++ version` to the term `version defined in toplevel CMakeLists.txt`, but it would be nice to directly understand which value is used with `--fix`

andistorm replied:
Maybe the hint should suggest using `--update-version` with the intended version instead

hikinggrass replied:
I've renamed it to _everest-framewok version_ and added an additional hint

hikinggrass committed:
Rename C++ version to "everest-framework version"

Add output on a version mismatch explaining that "--update-version" can be used to set a new version

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

andistorm reviewed (approved):
🐧
